.github/workflows/docs.yml

```diff
@@ -1,50 +1,80 @@
-name: Build Docs
+name: Build Documentation
 
-# We don't want pushes (or PRs) to gh-pages to kick anything off
 on:
+  workflow_dispatch:
   push:
-    branches: [ main ]
+    branches: [main]
+    paths:
+      - tests/*
+      - hydromt_wflow/*
+      - docs/*
+      - examples/*
+      - pyproject.toml
   pull_request:
-    branches: [ main ]
+    branches: [main]
+    paths:
+      - tests/*
+      - hydromt_wflow/*
+      - docs/*
+      - examples/*
+      - pyproject.toml
 
 jobs:
   # Build docs on Linux
   Docs:
-    name: linux docs
-    runs-on: ubuntu-latest
-    env:
-      DOC_VERSION: dev
     defaults:
       run:
         shell: bash -l {0}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest" ] #, "macos-latest", "windows-latest"]
+        python-version: ['3.11']
+        include:
+        - os: ubuntu-latest
+          label: linux-64
+          prefix: /usr/share/miniconda3/envs/hydromt_wflow
+
+    name: ${{ matrix.label }}
+    runs-on: ${{ matrix.os }}
```

Comment:
if you only test for one version an done os a matrix here is not necessary, but ti does make it easier to add those down the line. up to you whether you want to keep it.

Reply:
I like the fact that it gives me some variables to work with. but en general you're right.